hadoop-tools/hadoop-aws/src/test/java/org/apache/hadoop/fs/s3a/audit/TestS3AAuditLogMerger.java

```diff
@@ -0,0 +1,131 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ *  or more contributor license agreements.  See the NOTICE file
+ *  distributed with this work for additional information
+ *  regarding copyright ownership.  The ASF licenses this file
+ *  to you under the Apache License, Version 2.0 (the
+ *  "License"); you may not use this file except in compliance
+ *  with the License.  You may obtain a copy of the License at
+ *
+ *       http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ */
+
+package org.apache.hadoop.fs.s3a.audit;
+
+import java.io.File;
+import java.io.FileWriter;
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Paths;
+
+import org.junit.After;
+import org.junit.Before;
+import org.junit.Test;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import static org.junit.Assert.assertFalse;
+import static org.junit.Assert.assertTrue;
+
+/**
+ * MergerTest will implement different tests on Merger class methods.
+ */
+public class TestS3AAuditLogMerger {
+
+  private final Logger logger = LoggerFactory.getLogger(TestS3AAuditLogMerger.class);
+
+  private final S3AAuditLogMerger s3AAuditLogMerger = new S3AAuditLogMerger();
+
+  /**
+   * sample directories and files to test.
+   */
+  private final File auditLogFile = new File("AuditLogFile");
+  private final File sampleDirectory = new File("sampleFilesDirectory");
+  private final File emptyDirectory = new File("emptyFilesDirectory");
+  private final File firstSampleFile =
+      new File("sampleFilesDirectory", "sampleFile1.txt");
+  private final File secondSampleFile =
+      new File("sampleFilesDirectory", "sampleFile2.txt");
+  private final File thirdSampleFile =
+      new File("sampleFilesDirectory", "sampleFile3.txt");
```

Comment:
I think it'll be better to move these files into specific tests that need them and remove the deletion of them from teardown(). I think you won't have to worry about cleanup if you create a local file inside the test. Lookup `File.createTempFile()`, these are automatically deleted, check if you could use them in your tests.

Reply:
yeah i'll will try in this way then.